source/components/cardContainer/filters/cardContainerFilters.ts

```diff
@@ -0,0 +1,66 @@
+// /// <reference path='../../../typings/node/node.d.ts' />
+
+'use strict';
+
+import * as angular from 'angular';
+import * as _ from 'lodash';
+
+import { filters } from 'typescript-angular-utilities';
+import { IDataSource } from '../dataSources/dataSource';
+import { ITemplateObject } from '../../templateRenderer/templateRenderer';
+
+export let moduleName: string = 'rl.ui.components.cardContainer.filters.cardContainerFilters';
+export let componentName: string = 'rlCardContainerFilters';
+export let controllerName: string = 'CardContainerFiltersController';
+
+export interface IRenderableFilter extends filters.IFilter {
+	template: string | ITemplateObject;
+}
+
+export interface IFilterScope extends angular.IScope {
+	filter: filters.IFilter;
+	dataSource: IDataSource<any>;
+}
+
+export interface ICardContainerFiltersBindings {
+	filters: filters.IFilter[];
+	source: IDataSource<any>;
+}
+
+export class CardContainerFiltersController implements ICardContainerFiltersBindings {
+	filters: filters.IFilter[];
+	source: IDataSource<any>;
+	renderableFilters: IRenderableFilter[];
+
+	static $inject: string[] = ['$rootScope'];
+	constructor(private $rootScope: angular.IRootScopeService) { }
+
+	$onInit(): void {
+		this.renderableFilters = _(this.filters).filter((filter: IRenderableFilter): boolean => {
+			return filter.template != null;
+		}).map((filter: IRenderableFilter): IRenderableFilter => {
+			let scope: IFilterScope = <IFilterScope>this.$rootScope.$new();
```

Comment:
Creating this new scope from the root scope could be an interesting issue, but it's hard to tell what else could really be used here. Could be passed in from somewhere externally, or could derive from the scope of this directive.